src/platform/cert_capi.c

```diff
@@ -697,6 +697,177 @@ CxPlatCertParseChain(
     return (QUIC_CERTIFICATE*)LeafCertCtx;
 }
 
+_Success_(return != 0)
+QUIC_STATUS
+CxPlatGetPortableCertificate(
+    _In_ QUIC_CERTIFICATE* Certificate,
+    _Out_ QUIC_BUFFER* PortableCertificate,
+    _Out_ QUIC_BUFFER* CertificateChain
+    )
+{
+    QUIC_STATUS Status;
+    DWORD LastError;
+    CERT_CHAIN_PARA ChainPara;
+    CERT_ENHKEY_USAGE EnhKeyUsage;
+    CERT_USAGE_MATCH CertUsage;
+    PCCERT_CHAIN_CONTEXT ChainContext;
+    PCCERT_CONTEXT CertCtx = (PCCERT_CONTEXT)Certificate;
+    HCERTSTORE TempCertStore = NULL;
+    CERT_BLOB Blob = {0};
+
+    PortableCertificate->Buffer = CertCtx->pbCertEncoded;
```

Comment:
This is a little dangerous because if the `QUIC_CERTIFICATE` goes away before the caller is done with `PortableCertificate`, then this is invalid memory.
I know that's not how it's used today, but if it's used elsewhere in the future, this implementation detail will become a bug.

Reply:
I can use CertDuplicateCertificateContext to bump the ref count. I'll do that.

Reply:
you'll need to update `CxPlatFreePortableCertificate` to also take a `QUIC_CERTIFICATE` parameter do you can decrement the ref count, but that sounds like a good plan. The only alternative I can think of is an alloc+copy, which would waste memory.

Reply:
I made a wrapper struct to store the 3 items.